src/cmd/hotspots/transfer.rs

```diff
@@ -0,0 +1,150 @@
+};
+use std::io;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+/// Transfer hotspot as buyer or seller.
+pub enum Transfer {
+    /// Create and sign transaction to sell a hotspot, outputting it as base64 for counter-party
+    Sell(Sell),
+    /// Ingest a transaction to buy a hotspot from base64.
+    /// Signs and submits the transaction to the API
+    Buy(Buy),
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Sell {
+    /// Public address of gateway to be transferred
+    gateway: String,
+    /// The recipient of the gateway transfer
+    buyer: String,
+    /// Price in HNT to be paid by recipient of transfer
+    price: Option<Hnt>,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Buy {
+    /// Base64 encoded transaction to sign. If no transaction if given
+    /// stdin is read for the transaction. Note that the stdin feature
+    /// only works if the wallet password is set in the
+    /// HELIUM_WALLET_PASSWORD environment variable
+    #[structopt(name = "TRANSACTION")]
+    txn: Option<String>,
+    #[structopt(long)]
+    /// Commit to sign and submit the transaction
+    commit: bool,
+}
+
+impl Buy {
+    fn read_txn(&self) -> Result<String> {
+        match &self.txn {
+            Some(txn) => Ok(txn.to_string()),
+            None => {
+                let mut buffer = String::new();
+                io::stdin().read_line(&mut buffer)?;
+                Ok(buffer.trim().to_string())
+            }
+        }
+    }
+}
+
+impl Transfer {
+    pub fn run(self, opts: Opts) -> Result {
+        let wallet = load_wallet(opts.files)?;
+        let client = Client::new_with_base_url(api_url());
+
+        match self {
+            Self::Sell(sell) => {
+                let seller = wallet.pubkey_bin.to_vec();
+                let buyer = PubKeyBin::from_b58(&sell.buyer)?;
+                let buyer_account = client.get_account(&buyer.to_b58()?)?;
+                let gateway = PubKeyBin::from_b58(&sell.gateway)?.to_vec();
+
+                let mut txn = BlockchainTxnTransferHotspotV1 {
+                    fee: 0,
+                    seller,
+                    gateway,
+                    buyer: buyer.to_vec(),
+                    seller_signature: vec![],
+                    buyer_signature: vec![],
+                    amount_to_seller: if let Some(price) = sell.price {
+                        price.to_bones()
+                    } else {
+                        0
+                    },
+                    buyer_nonce: buyer_account.speculative_nonce + 1,
+                };
+                txn.fee = txn.txn_fee(&get_txn_fees(&client)?)?;
+                eprintln!("{:#?}", txn.to_json()?);
+
+                let password = get_password(false)?;
+                let keypair = wallet.decrypt(password.as_bytes())?;
+                txn.seller_signature = txn.sign(&keypair)?;
+                eprintln!("{}", txn.in_envelope().to_b64()?);
```

Comment:
Still not using print_json and you definitely shouldn't use eprintln! in this case

Reply:
This part isn't JSON and just shoots out the b64 of the txn the same as the Miner does for add hotspot. But eprint ain't right. Fixed it